testsuite/MDAnalysisTests/lib/test_cutil.py

```diff
@@ -45,6 +52,10 @@ def test_unique_int_1d(values):
     assert type(res) == type(ref)
     assert res.dtype == ref.dtype
 
+def test_array_unique_int_1d():
+    universe=u()
+    assert unique_int_1d(universe.atoms[100:130].resindices) == [5,6,7]
```

Comment:
For array comparisons, use the numpy functions, like
```suggestion
    assert_equal(unique_int_1d(universe.atoms[100:130].resindices),  [5,6,7])
```
(Make sure that `assert_equal` is imported from `numpy.testing`)

Reply:
`universe.atoms[100:130].resindices` should be replaced with something like `np.array([6, 5, 5, 7], dtype=np.int32)` or something like that – check what you get on your system  as the dtype of `universe.atoms[100:130].resindices.dtype`.